src/App/Fossa/Config/Common.hs

```diff
@@ -302,17 +308,36 @@ collectRevisionData (BaseDir basedir) maybeConfig cacheStrategy cliOverride = do
   let override = collectRevisionOverride maybeConfig cliOverride
   case cacheStrategy of
     ReadOnly -> do
-      inferred <- inferProjectFromVCS basedir <||> inferProjectCached basedir <||> inferProjectDefault basedir
+      inferred <- inferVCSInfo $ inferProjectCached basedir <||> inferProjectDefault basedir
       pure $ mergeOverride override inferred
     WriteOnly -> do
-      inferred <- inferProjectFromVCS basedir <||> inferProjectDefault basedir
+      inferred <- inferVCSInfo $ inferProjectDefault basedir
       let revision = mergeOverride override inferred
       saveRevision revision
       pure revision
+  where
+    inferVCSInfo ::
```

Comment:
decided to put this function under a 'where' clause as opposed to having it in a 'let'  bc I felt like it was a little bit neater / readable